### ISSUE TITLE
boot-cljs eats compile error

### ISSUE THREAD

**Issue by FossiFoo** [boot-clj/boot-cljs#150]
after getting no output on upgrading cljs to 1.9.473, i tracked an issue down to the "geheimnis" library.

if i build including it, the js doesn't get written: boot -d adzerk/boot-cljs:1.7.228-2 -d org.clojure/clojurescript:1.9.473 -d io.replikativ/geheimnis:0.1.0 -BP -s foo cljs target

with "foo" being a simple core.cljs not even requiring the package
I've made a small poc at https://github.com/FossiFoo/cljs-compile-poc

lein reveals the error in some reader macro.

also see https://github.com/replikativ/geheimnis/issues/1

**Comment by Deraen** [comment]
2.0.0-SNAPSHOT also shows the exception:

```
~/tmp/cljs-compile-poc master*
❯ boot cljs target
Classpath conflict: org.clojure/clojure version 1.8.0 already loaded, NOT loading version 1.9.0-alpha14
Writing main.cljs.edn...
Compiling ClojureScript...
• main.js
                                       java.lang.Thread.run                  Thread.java:  745
         java.util.concurrent.ThreadPoolExecutor$Worker.run      ThreadPoolExecutor.java:  617
          java.util.concurrent.ThreadPoolExecutor.runWorker      ThreadPoolExecutor.java: 1142
                        java.util.concurrent.FutureTask.run              FutureTask.java:  266
                                                        ...                                   
                        clojure.core/binding-conveyor-fn/fn                     core.clj: 1938
                              adzerk.boot-cljs/compile-1/fn                boot_cljs.clj:  128
                                   adzerk.boot-cljs/compile                boot_cljs.clj:   71
                                          boot.pod/call-in*                      pod.clj:  413
                                                        ...                                   
org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke  ClojureRuntimeShimImpl.java:  102
org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke  ClojureRuntimeShimImpl.java:  109
                                                        ...                                   
                                          boot.pod/call-in*                      pod.clj:  410
                                      boot.pod/eval-fn-call                      pod.clj:  359
                                         clojure.core/apply                     core.clj:  657
                                                        ...                                   
                         adzerk.boot-cljs.impl/compile-cljs                     impl.clj:  109
                                       cljs.build.api/build                      api.clj:  206
                                         cljs.closure/build                  closure.clj: 2116
                            cljs.closure/load-data-readers!                  closure.clj: 2048
                                                        ...                                   
                                    clojure.core/memoize/fn                     core.clj: 6235
                                         clojure.core/apply                     core.clj:  657
                                                        ...                                   
                             cljs.closure/get-data-readers*                  closure.clj: 2040
                             cljs.closure/get-data-readers*                  closure.clj: 2043
                                        clojure.core/reduce                     core.clj: 6704
                                clojure.core.protocols/fn/G                protocols.clj:   13
                                  clojure.core.protocols/fn                protocols.clj:   75
                          clojure.core.protocols/seq-reduce                protocols.clj:   31
                                clojure.core.protocols/fn/G                protocols.clj:   19
                                  clojure.core.protocols/fn                protocols.clj:  167
                         cljs.closure/load-data-reader-file                  closure.clj: 2018
                                       clojure.core/ex-info                     core.clj: 4725
clojure.lang.ExceptionInfo: Not a valid data-reader map
    url: #object[java.net.URL 0x2bcdf83b "jar:file:/home/juho/.m2/repository/io/replikativ/geheimnis/0.1.0/geheimnis-0.1.0.jar!/data_readers.cljc"]
clojure.lang.ExceptionInfo: Not a valid data-reader map
    from: :boot-cljs
     url: #object[java.net.URL 0x2bcdf83b "jar:file:/home/juho/.m2/repository/io/replikativ/geheimnis/0.1.0/geheimnis-0.1.0.jar!/data_readers.cljc"]
clojure.lang.ExceptionInfo: Not a valid data-reader map
    file: "/tmp/boot.user8032194686550077065.clj"
    line: 23

```

**Comment by FossiFoo** [comment]
Oh, okay, weird. When I tried it, I also got nothing, but maybe I did that
wrong.

**Comment by Deraen** [comment]
Should be fixed in 2.0.0.
